src/gen.ts

```diff
@@ -136,9 +136,21 @@ export function generateLoaderOptions(
   genCode += `${Object.entries(options).map(([rootKey, rootValue]) => {
     if (rootKey === 'nuxtI18nOptions') {
       let genCodes = `export const resolveNuxtI18nOptions = async (context) => {\n`
-      genCodes += `  const ${rootKey} = ${JSON.stringify(rootValue || {})}\n`
-      if (vueI18nPath) {
-        genCodes += `${rootKey}.vueI18n = vueI18nOptions\n`
+      genCodes += `  const ${rootKey} = Object({})\n`
+      for (const [key, value] of Object.entries(rootValue)) {
+        if (key === 'vueI18n') {
+          // const optionLoaderVariable = `${key}OptionsLoader`
+          // genCodes += `  const ${optionLoaderVariable} = ${isString(value) && value !== ''
+          //     // ? `async (context) => import(${toCode(value)}).then(r => (r.default || r)(context))\n`
+          //     ? `async (context) => import(${toCode(value)}).then(r => (r.default || r))\n`
+          //     : `async (context) => ${toCode({})}\n`
+          // }`
+          // genCodes += `  ${rootKey}.${key} = await ${optionLoaderVariable}(context)\n`
+          genCodes += `  ${rootKey}.${key} = ${vueI18nConfigPath != null ? 'vueI18nOptions' : {} }\n`
+          genCodes += `  if (${rootKey}.${key}.messages) { console.warn("[${NUXT_I18N_MODULE_ID}]: Cannot include 'messages' option in '${value}'. Please use Lazy-load translations."); ${rootKey}.${key}.messages = {}; }\n`
```

Comment:
MENO(TODO):
If we have `messages` options, we must pre-compile the resource using dynamic import & bundler (virtual module). This is because, in production, vue-i18n assumes that the resource is pre-compiled.